src/compiler/sys.ts

```diff
@@ -2,6 +2,16 @@ declare function setTimeout(handler: (...args: any[]) => void, timeout: number):
 declare function clearTimeout(handle: any): void;
 
 namespace ts {
+    /**
+     * djb2 hashing algorithm
+     * http://www.cse.yorku.ca/~oz/hash.html
+     */
+    /* @internal */
+    export function generateDjb2Hash(data: string): string {
+        const chars = data.split("").map(str => str.charCodeAt(0));

```

Comment:
Is this used a lot? It seems like a for loop over the character indices would allocate less than data.split.map. Of course the optimizer might do the right thing.

Reply:
Later: oh, this is just a code move.

Reply:
Note that this is a fallback for creating hash to store instead of storing text of generated .d.ts files signature. With this change it would also be used as backup (when node api doesn't support creating hash) to get the hash for source File text as sourceFile's version

Reply:
Seems expensive for what it does, but not worth changing right now I think.

Reply:
Yeah thats the default function that is independent of this PR.